src/commands/package_engines.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+
+const getContents = require('../../lib/get-contents');
+
+exports.command = 'package:engines [search]';
+exports.desc = 'search for a string within the `package.json` engines field';
+
+const processJson = (content) => {  
+    return JSON.parse(content);
+};
+
+exports.handler = function (argv) {
+    const { token, search } = argv;
+    const repositories = fs.readFileSync('/dev/stdin').toString().split("\n");
+    const path = 'package.json';
+
+    const getPackageJson = getContents({
+        githubToken: token,
+        path
+    });
+    const allRepos = repositories.map(
+        repository => getPackageJson(repository)
+            .then((processJson))
+            .catch(() => {
+                console.error(`JSON PARSE ERROR: ${path} parse error in '${repository}'`)
+            })
+            .then((json = {}) => {
+                const { engines } = json;
+                if (engines) {
+                    const enginesOutput = Object.keys(engines).reduce((prevEngineName, engineName, curIndex) => {
+                        const prev = !!curIndex ? `${prevEngineName}@${engines[prevEngineName]}\t` : '';
+                        return  `${prev}${engineName}@${engines[engineName]}`
```

Comment:
I think there's too much going on in this reduce function for it to be easily readable, I think it would be good to make it clearer what's happening even if that means more code!

Reply:
Hope https://github.com/Financial-Times/github-repositories-contents-search/pull/12/files#diff-09f08511e6c1224ba0dc3a84d1601658R15 makes it more readable. Otherwise, I can give it another go :)

Reply:
How about something like this? 😊 

```
const getEnginesString = engines => {
  let myArray = [];
  for (const name in engines) {
    myArray.push(`${name}@${engines[name]}`);
  }
  return myArray.join("\t");
}
```

Reply:
Great feedback 👍 

Have updated to use map, so simplify it even more:

```
const enginesReport = engines => {
    return Object.keys(engines)
        .map(name => `${name}@${engines[name]}`)
        .join("\t");
};
```